src/main/java/cn/nukkit/block/BlockRedFlower.java

```diff
@@ -1,21 +1,108 @@
-    public static final BlockProperties PROPERTIES = new BlockProperties("minecraft:red_flower", CommonBlockProperties.FLOWER_TYPE);
+import java.util.concurrent.ThreadLocalRandom;
 
-    @Override
-    public @NotNull BlockProperties getProperties() {
-        return PROPERTIES;
-    }
+public class BlockRedFlower extends BlockFlowable implements BlockFlowerPot.FlowerPotBlock {
+    public static final BlockProperties PROPERTIES = new BlockProperties(RED_FLOWER, CommonBlockProperties.FLOWER_TYPE);
 
     public BlockRedFlower() {
-        this(PROPERTIES.getDefaultState());
+        super(PROPERTIES.getDefaultState());
     }
 
     public BlockRedFlower(BlockState blockstate) {
         super(blockstate);
     }
-}
\ No newline at end of file
+
+    @NotNull
+    @Override
+    public BlockProperties getProperties() {
+        return PROPERTIES;
+    }
+
+    public FlowerType getFlowerType() {
+        return getPropertyValue(CommonBlockProperties.FLOWER_TYPE);
+    }
+
+    public void setFlowerType(FlowerType flowerType) {
+        setPropertyValue(CommonBlockProperties.FLOWER_TYPE, flowerType);
+    }
+
+
+    public static boolean isSupportValid(Block block) {
+        return switch (block.getId()) {
+            case GRASS, DIRT, FARMLAND, PODZOL, DIRT_WITH_ROOTS, MOSS_BLOCK -> true;
+            default -> false;
+        };
+    }
+
+
+    public boolean canPlantOn(Block block) {
+        return isSupportValid(block);
+    }
+
+    @Override
+    public boolean place(@NotNull Item item, @NotNull Block block, @NotNull Block target, @NotNull BlockFace face, double fx, double fy, double fz, Player player) {
+        Block down = this.down();
+        if (canPlantOn(down)) {
+            this.getLevel().setBlock(block, this, true);
+
+            return true;
+        }
+        return false;
+    }
+
+
+    @Override
+    public int onUpdate(int type) {
+        if (type == Level.BLOCK_UPDATE_NORMAL) {
+            if (!canPlantOn(down())) {
+                this.getLevel().useBreakOn(this);
+
+                return Level.BLOCK_UPDATE_NORMAL;
+            }
+        }
+
+        return 0;
+    }
+
+    @Override
+    public boolean canBeActivated() {
+        return true;
+    }
+
+    @Override
+    public boolean onActivate(@NotNull Item item, Player player) {
+        if (item.isFertilizer()) { //Bone meal
+            if (player != null && (player.gamemode & 0x01) == 0) {
+                item.count--;
+            }
+
+            this.level.addParticle(new BoneMealParticle(this));
+
+            for (int i = 0; i < 8; i++) {
+                Vector3 vec = this.add(
+                        ThreadLocalRandom.current().nextInt(-3, 4),
+                        ThreadLocalRandom.current().nextInt(-1, 2),
+                        ThreadLocalRandom.current().nextInt(-3, 4));
+
+                if (level.getBlock(vec).getId() == AIR && level.getBlock(vec.down()).getId() == GRASS && vec.getY() >= 0 && vec.getY() < level.getMaxHeight()) {
```

Comment:
![image](https://github.com/PowerNukkitX/PowerNukkitX/assets/69153398/49bb1f7b-2298-488d-8a56-3816e4fa63ee)
Fix this bug to get the world min and max height from getDimensionData for Level. Also you removed `getUncommonFlower`, please keep it